src/ansi-c/expr2c.cpp

```diff
@@ -915,8 +915,8 @@ std::string expr2ct::convert_update(
   std::string op0, op1, op2;
   unsigned p0, p2;
 
-  op0=convert_with_precedence(src.op0(), p0);
-  op2=convert_with_precedence(src.op2(), p2);
+  op0 = convert_with_precedence(to_update_expr(src).op0(), p0);
+  op2 = convert_with_precedence(to_update_expr(src).op2(), p2);
```

Comment:
Do `to_update_expr` just once for the entire method, or maybe even make `update_exprt` the type of the `src` parameter?